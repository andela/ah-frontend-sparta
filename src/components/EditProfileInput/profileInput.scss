@import '../../assets/scss/shared.scss';

.profileInputContainer {
  margin-bottom: rem(15);

  .profileInput {
    font-size: rem(15);
    padding: rem(5);
    border: rem(1) solid #283560;
    border-radius: rem(5);
  }
}
