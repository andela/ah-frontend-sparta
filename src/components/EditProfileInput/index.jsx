import React from 'react';
import PropTypes from 'prop-types';

import './profileInput.scss';

const EditProfileInput = ({
  label, id, type, value, onChange, disabled,
}) => (
  <div className="profileInputContainer">
    <div className="label">
      <span>{`${label}:`}</span>
    </div>
    <input
      className="col-6 profileInput"
      id={id}
      type={type}
      defaultValue={value}
      onChange={onChange}
      disabled={disabled === true}
    />
  </div>
);

EditProfileInput.propTypes = {
  label: PropTypes.string,
  id: PropTypes.string,
  type: PropTypes.string,
  value: PropTypes.string,
  disabled: PropTypes.bool,
  onChange: PropTypes.func,
};

EditProfileInput.defaultProps = {
  label: '',
  id: '',
  type: '',
  value: '',
  disabled: false,
  onChange: () => {},
};

export default EditProfileInput;
