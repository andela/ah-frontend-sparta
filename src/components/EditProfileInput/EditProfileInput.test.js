import React from 'react';
import renderer from 'react-test-renderer';
import EditProfileInput from './index';

test('snapshot test', () => {
  const wrapper = renderer.create(<EditProfileInput />);
  expect(wrapper).toMatchSnapshot();
});
