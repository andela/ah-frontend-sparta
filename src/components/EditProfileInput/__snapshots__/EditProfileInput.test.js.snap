// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot test 1`] = `
<div
  className="profileInputContainer"
>
  <div
    className="label"
  >
    <span>
      :
    </span>
  </div>
  <input
    className="col-6 profileInput"
    defaultValue=""
    disabled={false}
    id=""
    onChange={[Function]}
    type=""
  />
</div>
`;
